packages/swap/src/deposit-tracking/index.ts

```diff
@@ -1,19 +1,40 @@
-import { Chain, Chains } from '@/shared/enums';
+import { Asset, Chain, Chains } from '@/shared/enums';
+import RedisClient from '@/shared/node-apis/redis';
 import { fetchPendingBitcoinDeposit } from './bitcoin';
+import prisma from '../client';
+import env from '../config/env';
+
+const redis = env.REDIS_URL ? new RedisClient(env.REDIS_URL) : undefined;
 
 export type PendingDeposit = {
   amount: string;
-  transactionHash: string;
+  transactionHash?: string;
   transactionConfirmations: number;
 };
 
 export const getPendingDeposit = async (
   chain: Chain,
+  asset: Asset,
   address: string,
 ): Promise<PendingDeposit | undefined> => {
   if (chain === Chains.Bitcoin) {
     return fetchPendingBitcoinDeposit(address);
   }
 
-  return undefined;
+  if (!redis) return undefined;
+
+  const [deposits, tracking] = await Promise.all([
+    redis.getDeposits(chain, asset, address),
+    prisma.chainTracking.findFirst({ where: { chain } }),
+  ]);
+
+  if (deposits.length === 0 || tracking == null) return undefined;
+
+  const confirmations =
+    tracking.height - BigInt(deposits[0].deposit_chain_block_height);
+
+  return {
+    amount: deposits[0].amount.toString(),
```

Comment:
what if there are multiple deposits with the same address?

Reply:
It's an edge case that I'm not sure there's a good way of handling. For the purposes of our swapping app, this oldest deposit will be the one that will cause the app to progress forward in the flow.

We need to refactor our deposit channel status flow to return multiple swaps and stuff. I think at that point we could handle the case of multiple deposits but right now I can't think of much we can really do